### DISCUSSION TITLE
Different radius

### DISCUSSION THREAD

**Discussion by hosseinrezaa** [InteractiveComputerGraphics/SPlisHSPlasH#144] [Q&A]
Hi
1- Is it possible to have a simulation with a different particle radius?
Regards

**Comment by janbender** [comment]
Currently, this is not possible. Using different radii in the simulation would require a special neighborhood search and an adaption of almost all equations in the different solvers.

**Reply by hosseinrezaa** [reply]
Possible to consider roughness on solid surfaces?

**Reply by janbender** [reply]
You could implement a friction model like the one proposed here: 
https://animation.rwth-aachen.de/publication/0554/

**Reply by kaiserm99** [reply]
Ist this possible by now ? If so, how should one modify the code to accommodate this.

Thx, Marco :)

**Reply by janbender** [reply]
Yes, SPlisHSPlasH has a modular structure, so it is quite simple to implement a new non-pressure force.
